Code/src/main/java/nl/utwente/viskell/ghcj/GhciSession.java

```diff
@@ -37,7 +37,7 @@
     private Evaluator ghci;
 
     /** Gets filled with a HaskellCatalog instance when ghci is ready. */
-    private HaskellCatalog catalog;
+    private static HaskellCatalog catalog;
```

Comment:
If that's the case, then we will need bigger changes.... as we need to get "a"/"the" catalog from the static fromBundleFragment() method way down the call stack....so the deserialization (starting at toplevel pane or menu actions) will need to know *which* session to use to deserialize with, and pass it down all the fromBundle() & fromBundleFragment() calls to where it is needed.